packages/components/Switch/src/useSwitch.ts

```diff
@@ -0,0 +1,94 @@
+  const {
+    onChange,
+    checked,
+    defaultChecked,
+    accessibilityRole,
+    accessibilityLabel,
+    onAccessibilityTap,
+    onAccessibilityAction,
+    label,
+    labelPosition,
+    componentRef = defaultComponentRef,
+    disabled,
+    ...rest
+  } = props;
+
+  const isDisabled = !!disabled;
+  const initialCheckedState = !!(checked ?? defaultChecked);
+  const [checkedState, setCheckedState] = React.useState(initialCheckedState);
+  const focusRef = isDisabled ? null : componentRef;
+
+  const toggleCallback = React.useCallback(
+    (e: InteractionEvent) => {
+      const newCheckedState = checked !== undefined ? checked : !checkedState;
+      onChange && onChange(newCheckedState, e);
+      LayoutAnimation.configureNext(LayoutAnimation.create(200, LayoutAnimation.Types.linear, LayoutAnimation.Properties.scaleX));
+      setCheckedState(newCheckedState);
+    },
+    [onChange, setCheckedState, checkedState, checked],
+  );
+
+  const onClickWithFocus = useOnPressWithFocus(focusRef, toggleCallback);
+  const pressable = useAsPressable({ ...rest, disabled: isDisabled, onPress: onClickWithFocus });
+  const onKeyUpProps = useKeyProps(toggleCallback, ' ', 'Enter');
+  const currentCheckedState = checked ?? checkedState;
+
+  const accessibilityState: AccessibilityState = {
+    checked: props.accessibilityState?.checked ?? currentCheckedState,
+  };
+
+  const accessibilityActionsProp: AccessibilityActionInfo[] = [{ name: 'Toggle' }];
+
+  const onAccessibilityActionProp = React.useCallback(
+    (event: AccessibilityActionEvent) => {
+      switch (event.nativeEvent.actionName) {
+        case 'Toggle':
+          toggleCallback(event); // will likely replace this with something else
+          break;
+      }
+      onAccessibilityAction && onAccessibilityAction(event);
+    },
+    [toggleCallback, onAccessibilityAction],
+  );
+
+  return {
+    props: {
+      ...pressable.props,
+      accessible: true,
+      accessibilityRole: accessibilityRole ?? 'switch',
+      onAccessibilityTap: onAccessibilityTap ?? onChange,
+      accessibilityLabel: accessibilityLabel ?? label,
+      accessibilityState: getAccessibilityState(isDisabled, accessibilityState),
+      accessibilityActions: accessibilityActionsProp,
+      onAccessibilityAction: onAccessibilityActionProp,
+      focusable: !isDisabled,
+      ref: useViewCommandFocus(componentRef),
+      ...onKeyUpProps,
+      checked: currentCheckedState,
+      label: label,
+      labelPosition: labelPosition ?? 'after',
+    },
+    state: {
+      ...pressable.state,
+      toggleOn: currentCheckedState == true,
+      toggleOff: currentCheckedState == false,
+    },
+  };
+};
+
+const getAccessibilityState = memoize(getAccessibilityStateWorker);
+function getAccessibilityStateWorker(disabled: boolean, accessibilityState?: AccessibilityState) {
+  if (accessibilityState) {
+    return { disabled: disabled, ...accessibilityState };
+  }
+  return { disabled: disabled };
+}
```

Comment:
Can ignore if addressing accessibility in another PR - I would add `checked: boolean` as one of the parameters for `getAccessibilityStateWorker` since it's an expected state of this control and set it equal to `checkedState` on line 70 and remove the `accessibilityState` prop you create on line 45.

Reply:
Yeah I've removed accessibility changes from this PR and moved it to another branch which I will do a PR for later.